### ISSUE TITLE
Exception when authenticating

### ISSUE THREAD

**Issue by kezakez** [LedgerHQ/ledger-dotnet-api#3]
I am trying to use the u2f API but I get command not allowed every 2nd call
eg.

```
            var appId = new AppId(Encoders.Hex.DecodeData("d2e42c173c857991d5e1b6c81f3e07cbb9d5f57431fe41997c9445c14ce61ec4"));
            var challenge = Encoders.Hex.DecodeData("e6425678fbd7d3d8e311fbfb1db8d26c37cf9f16ac81c95848998a76ce3d3768");
            U2FClient u2f = U2FClient.GetHIDU2F().First();


            // Accept registration
            var reg = u2f.Register(challenge, appId);


            // Accept registration
            try
            {
                u2f.Authenticate(challenge, appId, reg.KeyHandle);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                //            LedgerWallet.LedgerWalletException: Command not allowed: Conditions of use not satisfied
                //                at LedgerWallet.LedgerClientBase.Throw(Int32 sw)
                //            at LedgerWallet.LedgerClientBase.< ExchangeApdus > d__14.MoveNext()
                //                -- - End of stack trace from previous location where exception was thrown ---
                //                at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                //            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                //            at LedgerWallet.U2F.U2FClient.< AuthenticateAsync > d__10.MoveNext()
                //                -- - End of stack trace from previous location where exception was thrown ---
                //                at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                //            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                //            at LedgerWallet.U2F.U2FClient.Authenticate(Byte[] challenge, AppId applicationId, KeyHandle keyHandle, CancellationToken cancellationToken)
                //            at Ledger.Program.Main(String[] args) in C: \Users\Keza\Documents\Visual Studio 2017\Projects\Ledger\Ledger\Program.cs:line 28
            }

            var login = u2f.Authenticate(challenge, appId, reg.KeyHandle);

            Console.WriteLine(login);
            //            LedgerWallet.U2F.U2FAuthenticationResponse
            Console.WriteLine(login.UserPresence);
            //            False
            Console.ReadLine();
```

reading the u2f spec it seems to be the test of user presence returning false
what am I doing wrong?

**Comment by NicolasDorier** [comment]
Does this line works?

```
   var login = u2f.Authenticate(challenge, appId, reg.KeyHandle);
```

Also have you tried to clone this project and run the test in `U2FTests.cs` (in debug mode)

**Comment by kezakez** [comment]
yes that line works, but only after the previous attempt fails.

Running the U2FTests fails for me on the first register attempt:
Assert.Throws() Failure
Expected: typeof(System.OperationCanceledException)
Actual:   typeof(LedgerWallet.LedgerWalletException): Command not allowed : Conditions of use not satisfied

The version of my u2f program on my ledger is 1.2.1 if that helps

**Comment by NicolasDorier** [comment]
Thanks I will try today. It is likely a problem of the ledger, will check if I can reproduce with Yubico as well.

**Comment by NicolasDorier** [comment]
Ok so the error is on my part, I incorrectly assumed how U2F devices worked. While my code was fine for ledger before it was an unintended consequence of their implementation.

I will push a new version of this package later today to fix your issue.

**Comment by NicolasDorier** [comment]
Actually it was a bug I introduced by mistake, I just fixed it. Please update to version `1.0.1.32`, it will work.

**Comment by kezakez** [comment]
Wow that was quick. Yeah it is working better.
I am not sure if it is related or is a separate issue but when I call the non async Authenticate method and press the x on my ledger to deny the attempt, it continues to prompt again straight away and the authenticate method does not return.

**Comment by NicolasDorier** [comment]
@kezakez so this is because U2F does not have a concept of "deny attempt". This is why there is no deny button on [yubico keys](https://www.yubico.com/products/yubikey-hardware/).

Client application just wait for a confirmation until a timeout.
